controllers/imagejob/imagejob_controller.go

```diff
@@ -197,40 +214,40 @@ func (r *Reconciler) handleRunningJob(ctx context.Context, imageJob *eraserv1alp
 	}
 
 	imageJob.Status = eraserv1alpha1.ImageJobStatus{
-		Desired:   imageJob.Status.Desired,
-		Succeeded: success,
-		Failed:    failed,
-		Phase:     eraserv1alpha1.PhaseCompleted,
-	}
-	if err := r.updateJobStatus(ctx, imageJob); err != nil {
-		return err
+		Desired:     imageJob.Status.Desired,
+		Succeeded:   success,
+		Failed:      failed,
+		Phase:       eraserv1alpha1.PhaseCompleted,
+		DeleteAfter: after(time.Now(), *successDelDelaySeconds),
 	}
 
-	// transfer results from imageStatus objects to imageList
-	statusList := &eraserv1alpha1.ImageStatusList{}
-	err = r.List(ctx, statusList)
-	if err != nil {
-		return err
+	if float64(success/imageJob.Status.Desired) < *successeRatio {
+		log.Info("Marking job as failed", "success-ratio", *successeRatio, "actual ratio", success/imageJob.Status.Desired)
+		imageJob.Status.Phase = eraserv1alpha1.PhaseFailed
+		imageJob.Status.DeleteAfter = after(time.Now(), *errDelDelaySeconds)
 	}
 
-	var nodeResult []eraserv1alpha1.NodeResult
-
-	for i := range statusList.Items {
-		nodeResult = append(nodeResult, eraserv1alpha1.NodeResult{
-			Name:   statusList.Items[i].Result.Node,
-			Images: statusList.Items[i].Result.Results,
-		})
+	if err := r.updateJobStatus(ctx, imageJob); err != nil {
+		return err
 	}
 
 	imageList := &eraserv1alpha1.ImageList{}
 	err = r.Get(ctx, types.NamespacedName{Name: imageJob.Spec.ImageListName}, imageList)
 	if err != nil {
 		return err
 	}
-	if err := r.updateImageListStatus(ctx, nodeResult, imageList); err != nil {
-		return err
-	}
-	return nil
+
+	now := metav1.Now()
+	imageList.Status.Timestamp = &now
```

Comment:
I'm thinking we could put this in a list of statuses from the last... maybe 5 runs + the job name for each run... but maybe a separate PR.